File: C Programming/problems on recursion/problem13.c
/*
13.  Write a recursive program which accept number from user and return smallest digit 
	
		Input : 87983 
	
		Output : 3 
	
		Author :Milind Nalawade
	
		Date : 26/08/2021
	
		Language : C programming
	
*/

#include<stdio.h>

//Logic
int min=8,mod=0;
int Output(int no)
{
	if(no==0)
	{
		return min;
	}
	else
	{
		mod=no%10;
		if(min>mod)
		{
			min=mod;
		}
		no=no/10;
		Output(no);
	}
}	

//Entry point function
int main()
{
	int no=0,ret=0;
	printf("\n Input : ");
	scanf("%d",&no);
	ret=Output(no);
	printf("\n Output : %d\n",ret);
	return 0;
}	

File: Python Programming/problems on recursion/problem13.py
'''
13.  Write a recursive program which accept number from user and return smallest digit 
	
		Input : 87983 
	
		Output : 3 
	
		Author :Milind Nalawade
	
		Date : 26/08/2021
	
		Language : Python 3.9.0

'''
min=8
#====
#Logic
#====
def Output(no):
	global min
	if no==0:
		return min
	else:
		mod=no%10
		if(min>mod):
			min=mod;
		no=no//10
		Output(no)
		
#=============
#Entry point function
#=============
def main():
	global mul
	no=int(input("\n Input : "))
	Output(no)
	print("\n Output : {} \n".format(min))
	
#=====
#starter
#=====
if __name__=="__main__":
	main()